ksqldb-parser/src/main/java/io/confluent/ksql/parser/properties/with/CreateSourceAsProperties.java

```diff
@@ -105,6 +99,11 @@ public SerdeOptions getSerdeOptions() {
     return SerdeOptions.of(builder.build());
   }
 
+  public Optional<FormatInfo> getKeyFormatInfo() {
+    return Optional.ofNullable(props.getString(CommonCreateConfigs.KEY_FORMAT_PROPERTY))
+        .map(format -> FormatInfo.of(format, ImmutableMap.of()));
+  }
```

Comment:
This should also return non-empty if `FORMATS` is set, (same for value version).